consensus/src/chained_bft/persistent_storage.rs

```diff
@@ -186,23 +186,26 @@ impl<T: Payload> RecoveryData<T> {
         // sort by round to guarantee the topological order of parent <- child
         blocks.sort_by_key(Block::round);
         let root_from_consensus = {
-            let id_to_round: HashMap<_, _> = blocks
+            let id_to_id_and_round: HashMap<_, _> = blocks
                 .iter()
-                .map(|block| (block.id(), block.round()))
+                .map(|block| (block.id(), (block.id(), block.round())))
                 .collect();
-            let mut round_and_id = None;
-            for qc in quorum_certs.iter() {
-                if let Some(committed_block_id) = qc.committed_block_id() {
-                    if let Some(round) = id_to_round.get(&committed_block_id) {
-                        match round_and_id {
-                            Some((r, _)) if r > round => (),
-                            _ => round_and_id = Some((round, committed_block_id)),
-                        }
-                    }
+
+            // We extract the committed block id of the qc of minimal round
```

Comment:
Agree that equality should not happen anyway. It becomes really subjective at this point, but I don't see how the new variant is easier to read than the old one. Tbh the old one seems more straightforward to me.
I'll let @zekun000 decide about it because he's the author of the original piece :).

Reply:
Re-simplified, thanks for the pushback.